clients/unshaded/src/main/java/tachyon/client/file/FileOutStream.java

```diff
@@ -136,6 +137,22 @@ public void close() throws IOException {
       }
     }
 
+    canComplete = canComplete || storeToTachyon();
+
+    if (canComplete) {
```

Comment:
I think that case suggesting something is wrong (right?) and we may at least LOG.error this